bundles/org.openhab.binding.ws980wifi/src/main/java/org/openhab/binding/ws980wifi/internal/discovery/ws980wifi.java

```diff
@@ -0,0 +1,284 @@
+            this.tempDewPoint = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 13, 15)))) / 10;
+            this.tempWindChill = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 16, 18)))) / 10;
+            this.heatIndex = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 19, 21)))) / 10;
+            this.humidityInside = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 22, 23))));
+            this.humidityOutside = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 24, 25))));
+            this.pressureAbsolut = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 26, 28)))) * 10;
+            this.pressureRelative = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 29, 31)))) * 10;
+            this.windDirection = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 32, 34))));
+            this.windSpeed = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 35, 37)))) / 10;
+            this.windSpeedGust = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 38, 40)))) / 10;
+            this.rainLastHour = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 41, 45)))) / 10;
+            this.rainLastDay = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 46, 50)))) / 10;
+            this.rainLastWeek = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 51, 55)))) / 10;
+            this.rainLastMonth = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 56, 60)))) / 10;
+            this.rainLastYear = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 61, 65)))) / 10;
+            this.rainTotal = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 66, 70)))) / 10;
+            this.lightLevel = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 71, 75)))) / 10;
+            this.uvRaw = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 76, 78))));
+            this.uvIndex = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 79, 80))));
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private byte[] tcpipRequest(byte[] sendBytes) {
+        byte[] receive = {};
+        try (Socket aSocket = new Socket(wsHost, wsPort);
+                OutputStream outStream = aSocket.getOutputStream();
+                DataOutputStream mesgToServer = new DataOutputStream(outStream);
+                InputStream inStream = aSocket.getInputStream();
+                DataInputStream answerFromServer = new DataInputStream(inStream);) {
+            mesgToServer.write(sendBytes, 0, sendBytes.length);
+            receive = new byte[1024];
+            answerFromServer.read(receive);
+            return receive;
+        } catch (Exception e) {
```

Comment:
Unless `Exception` is actually thrown (I can't see it from the code alone, so it maybe valid here), you should not catch it. Either catch the specific exception or RuntimeExecption if you want to catch unknown exceptions.

Reply:
modified to IOException